### PR TITLE
add docstring for learning

### PR DESCRIPTION
**Pull request by SenWu**

### CODE REVIEW

File: src/fonduer/learning/classifier.py [hunk 1]
```diff
@@ -4,23 +4,49 @@
...
+    :type cardinality: int
+    :param name: Name of the model
+    :type name: str
+    """
 
     def __init__(self, cardinality=2, name=None):
         self.name = name or self.__class__.__name__
         self.cardinality = cardinality
 
     def marginals(self, X, batch_size=None, **kwargs):
```
Comment (lukehsiao) [inline src/fonduer/learning/classifier.py]: Are kwargs needed here? Can we enumerate what they are hiding?

File: src/fonduer/learning/classifier.py [hunk 2]
```diff
@@ -138,11 +178,21 @@ def error_analysis(
...
         return True
 
     def _preprocess_data(self, X, Y=None):
-        """Generic preprocessing function"""
+        """Generic preprocessing function
+
+        :param X: The input data of the model
+        :type X: pair with candidates and corresponding features
+        :param Y: The labels of input data
+        :type Y: list
```
Comment (lukehsiao) [inline src/fonduer/learning/classifier.py]: Annotate return values too.

```
:return: description
```

File: src/fonduer/learning/disc_learning.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ class NoiseAwareModel(Classifier, nn.Module):
     Generic NoiseAwareModel class.
 
     :param seed: Random seed which is passed into both numpy and PyTorch.
+    :type seed: int
```
Comment (lukehsiao) [inline src/fonduer/learning/disc_learning.py]: There are kwargs hiding init parameters here too. What are they? Are they needed?

File: src/fonduer/learning/disc_models/logistic_regression.py [hunk 4]
```diff
@@ -8,19 +8,21 @@
 
 class LogisticRegression(NoiseAwareModel):
```
Comment (lukehsiao) [inline src/fonduer/learning/disc_models/logistic_regression.py]: What are the kwargs?

```
fonduer.learning.LogisticRegression(seed=123, **kwargs)
```
can they be removed?

File: src/fonduer/learning/disc_models/lstm.py [hunk 5]
```diff
@@ -16,12 +16,14 @@
 
 class LSTM(NoiseAwareModel):
```
Comment (lukehsiao) [inline src/fonduer/learning/disc_models/lstm.py]: Same comment here. How does user initialize this?
```
class fonduer.learning.LSTM(seed=123, **kwargs)
```
What are kwargs? Can we remove or enumerate?

File: src/fonduer/learning/disc_models/logistic_regression.py (hunk 4 above)
Comment (lukehsiao) [reply src/fonduer/learning/disc_models/logistic_regression.py]: Same comment for SparseLogisticRegression too

```
class fonduer.learning.SparseLogisticRegression(seed=123, **kwargs)
```

**Comment by lukehsiao** [comment]
Basically, Core Learning Objects are pretty much ready, just want to eliminate kwargs wherever possible.

The Learning Utilities are still missing several docstrings on initilizations for classes and some function parameters.

File: src/fonduer/learning/classifier.py (hunk 1 above)
Comment (SenWu) [reply src/fonduer/learning/classifier.py]: Done!

File: src/fonduer/learning/classifier.py (hunk 2 above)
Comment (SenWu) [reply src/fonduer/learning/classifier.py]: Done!

File: src/fonduer/learning/disc_learning.py [hunk 6]
```diff
@@ -20,14 +20,19 @@ class NoiseAwareModel(Classifier, nn.Module):
     Generic NoiseAwareModel class.
```
[the remark this answers is not in the document]
Comment (SenWu) [reply src/fonduer/learning/disc_learning.py]: Good point! I will reconstruct with MTL support pretty soon which will have a different design with `train()`. Do you think we should do it now or later?

[the remark this answers is not in the document]
Comment (SenWu) [reply src/fonduer/learning/disc_learning.py]: Removed all `kargs` in `train()` and use `self.setting` to collect all model settings instead. Let user to set generic params in `train()` and model specific params in `.fonduer-config.yaml`

File: src/fonduer/learning/disc_models/logistic_regression.py [hunk 7]
```diff
@@ -4,23 +4,37 @@
 from scipy.sparse import issparse
```
Comment (lukehsiao) [inline src/fonduer/learning/disc_models/logistic_regression.py]: `get_train_idxs` takes `rand_state`, but it is not described in the docstring. You can add types to the parameters too.

Other than that LGTM.

Comment (SenWu) [reply src/fonduer/learning/disc_models/logistic_regression.py]: Done!

**Comment by SenWu** [comment]
Fixed! Merge to docstrings branch if LGTY. Thanks.

**Pull request merged**